packages/features/bookings/lib/handleNewBooking/test/team-bookings/collective-scheduling.test.ts

```diff
@@ -1509,7 +1509,128 @@ describe("handleNewBooking", () => {
       });
     });
 
-    test.todo("Round Robin booking");
+    describe("Round Robin Assignment", () => {
+      test(`successfully books contact owner if rr lead skip is enabled`, async ({ emails }) => {
+        const handleNewBooking = (await import("@calcom/features/bookings/lib/handleNewBooking")).default;
+        const booker = getBooker({
+          email: "booker@example.com",
+          name: "Booker",
+        });
+
+        const otherTeamMembers = [
+          {
+            name: "Other Team Member 1",
+            username: "other-team-member-1",
+            timeZone: Timezones["+5:30"],
+            defaultScheduleId: 1001,
+            email: "other-team-member-1@example.com",
+            id: 102,
+            schedules: [{ ...TestData.schedules.IstWorkHours, id: 1001 }],
+          },
+        ];
+
+        const organizer = getOrganizer({
+          name: "Organizer",
+          email: "organizer@example.com",
+          id: 101,
+          schedules: [TestData.schedules.IstWorkHours],
+        });
+
+        mockSuccessfulVideoMeetingCreation({
+          metadataLookupKey: appStoreMetadata.dailyvideo.dirName,
+          videoMeetingData: {
+            id: "MOCK_ID",
+            password: "MOCK_PASS",
+            url: `http://mock-dailyvideo.example.com/meeting-1`,
+          },
+        });
+
+        const { eventTypes } = await createBookingScenario(
+          getScenarioData({
+            webhooks: [
+              {
+                userId: organizer.id,
+                eventTriggers: ["BOOKING_CREATED"],
+                subscriberUrl: "http://my-webhook.example.com",
+                active: true,
+                eventTypeId: 1,
+                appId: null,
+              },
+            ],
+            eventTypes: [
+              {
+                id: 1,
+                slotInterval: 30,
+                schedulingType: SchedulingType.ROUND_ROBIN,
+                length: 30,
+                metadata: {
+                  apps: {
+                    salesforce: {
+                      enabled: true,
+                      appCategories: ["crm"],
+                      roundRobinLeadSkip: true,
+                    },
+                  },
+                },
+                users: [
+                  {
+                    id: 101,
+                  },
+                  {
+                    id: 102,
+                  },
+                ],
+              },
+            ],
+            organizer,
+            usersApartFromOrganizer: otherTeamMembers,
+          })
+        );
+
+        const bookingData = {
+          eventTypeId: 1,
+          teamMemberEmail: otherTeamMembers[0].email,
```

Comment:
It's not ideal that we pass the contact owner here to `handleNewBooking`. This means that the RR skip won't work when the booking is done via API. Also, sometimes when confirming the booking too fast after providing the booker's email, `teamMemberEmail` wasn't set yet and the RR skip was ignored